File: test/data/cpp/test04/StateConfiguring.cpp
#include "StateConfiguring.h"

StateConfiguring::StateConfiguring(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateConfiguring::~StateConfiguring()
{
}
BaseState* StateConfiguring::TransitEvConfig()
{
    _controllee->DoAction2();
    _stateController->InstanceOfIdle->Setup();
    return _stateController->InstanceOfIdle;
}
const char* StateConfiguring::GetStateName()
{
    return "NotShooting.Configuring";
}

File: test/data/typescript/test04/StateConfiguring.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
export class StateConfiguring extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public TransitEvConfig(): BaseState | null
    {
        this._controlleeImp.DoAction2();
        this._stateController.InstanceOfIdle.Setup(false, false);
        return this._stateController.InstanceOfIdle;
    }
    public  GetStateName(): string
    {
        return "NotShooting.Configuring";
    }
    public GetParent(): BaseState | null
    {
        return this._stateController.InstanceOfNotShooting;
    }
}